utils/serverStatistic.js

```diff
@@ -0,0 +1,130 @@
+    const data = await response.json();
+    return data.data;
+  } catch (error) {
+    console.error('Error:', error);
+  }
+};
+
+const getVehicleStats = async (accountId) => {
+  const result = {};
+  const vehicleStatsURL = `${BASE_URL}/tanks/stats/?application_id=${application_id}&account_id=${accountId}`;
+  const vehicleAchievmentsURL = `${BASE_URL}/tanks/achievements/?application_id=${application_id}&account_id=${accountId}`;
+  const vehicleStats = await getInfo(vehicleStatsURL);
+  const vehicleAchievments = await getInfo(vehicleAchievmentsURL);
+
+  const stats = vehicleStats[accountId];
+  const achievements = vehicleAchievments[accountId];
+
+  if (!stats || !achievements) return;
+
+  stats.map((tankStats) => {
+    const battleLifeTime = tankStats.battle_life_time;
+    if (!battleLifeTime) return;
+    const regular = getStatsObject(statsNames, tankStats.all);
+    result[tankStats.tank_id] = { regular, battleLifeTime };
+  });
+
+  achievements.map((tankAchivements) => {
+    if (!result[tankAchivements.tank_id]) return;
+    const mastery = getStatsObject(achievementsName, tankAchivements.achievements);
+    result[tankAchivements.tank_id].mastery = mastery;
+  });
+
+  return Object.entries(result).map(([wotId, data]) => ({ wotId, ...data }));
+};
+
+const fetchPlayerStats = async (playerIds) => {
+  const playerStatsURL = `${BASE_URL}/account/info/?application_id=${application_id}&account_id=${playerIds.join(',')}&extra=statistics.rating`;
+  const playerAchievmentsURL = `${BASE_URL}/account/achievements/?application_id=${application_id}&account_id=${playerIds.join(',')}`;
+
+  try {
+    const playerStats = await getInfo(playerStatsURL);
+    const playerAchievments = await getInfo(playerAchievmentsURL);
+
+    for (const accountId in playerStats) {
+      const stats = playerStats[accountId]?.statistics;
+      const achievements = playerAchievments[accountId]?.achievements;
+      const vehicleStats = await getVehicleStats(accountId);
+
+      if (!stats || !achievements) continue;
+      const regularStats = stats.all;
+      const ratingStats = stats.rating;
+
+      if (JSON.stringify(allPlayerStats.account) === '{}') {
+        allPlayerStats.account.regular = getStatsObject(statsNames, regularStats);
+        allPlayerStats.account.rating = getStatsObject(statsNames, ratingStats);
+        allPlayerStats.account.mastery = getStatsObject(achievementsName, achievements);
+      } else {
+        statsNames.map((statName) => {
+          allPlayerStats.account.regular[statName] = allPlayerStats.account.regular[statName] + regularStats[statName];
+          allPlayerStats.account.rating[statName] = allPlayerStats.account.rating[statName] + ratingStats[statName];
+        });
+        achievementsName.map((achievementName) => {
+          allPlayerStats.account.mastery[achievementName] = allPlayerStats.account.mastery[achievementName] + (achievements[achievementName] || 0);
+        });
+      }
```

Comment:
You can use short form to sum results.
```suggestion
        statsNames.map((statName) => {
          allPlayerStats.account.regular[statName] += regularStats[statName];
          allPlayerStats.account.rating[statName] += ratingStats[statName];
        });
        achievementsName.map((achievementName) => {
          allPlayerStats.account.mastery[achievementName] +=(achievements[achievementName] || 0);
        });
      }
```